### ISSUE TITLE
Parallel CpGrid test with aquifer cells is missing.

### ISSUE THREAD

**Issue by blattms** [OPM/opm-grid#668]
While looking at OPM/opm-simulators#740 I started wondering whether we do actually  distribute the information about aquifer cells when load balancing. I hope we do, but we should definitely add one test for that to the test suite. At least could not find one.

**Comment by bska** [comment]
> While looking at [OPM/opm-simulators#740](https://github.com/OPM/opm-simulators/pull/740)

Do you mean OPM/opm-simulators#4740?

> I started wondering whether we actually distribute the information about aquifer cells when load balancing.

Kind of.  There's nothing in the load balancing step that uses or is aware of numerical aquifers, but once we're running in distributed mode every process/rank's [collection of aquifers](https://github.com/OPM/opm-simulators/blob/10229a1781d3e055cc91a473fd84ed82eed3df41/opm/simulators/aquifers/BlackoilAquiferModel.hpp#L126-L127) knows about [all numerical aquifers in the model](https://github.com/OPM/opm-simulators/blob/10229a1781d3e055cc91a473fd84ed82eed3df41/opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp#L255-L262).

**Comment by blattms** [comment]
> > While looking at [OPM/opm-simulators#740](https://github.com/OPM/opm-simulators/pull/740)
> 
> Do you mean [OPM/opm-simulators#4740](https://github.com/OPM/opm-simulators/pull/4740)?
> 

Yes that is what I meant. In [AquiferGridUtils.hpp#L52](https://github.com/akva2/opm-simulators/blob/4d35ec26de78d3b988939128ad7817f179c54d29/opm/simulators/aquifers/AquiferGridUtils.hpp#L52) we ask CpGrid about the aquifer cells, but I cannot find that this information is actually ever transferred to the load balance grid. For the unbalanced grid the aquifer cells a added when processing the eclipse files it seems

**Comment by bska** [comment]
> While looking at  [OPM/opm-simulators#4740](https://github.com/OPM/opm-simulators/pull/4740) I started wondering whether we actually distribute the information about aquifer cells when load balancing. In [AquiferGridUtils.hpp#L52](https://github.com/akva2/opm-simulators/blob/4d35ec26de78d3b988939128ad7817f179c54d29/opm/simulators/aquifers/AquiferGridUtils.hpp#L52) we ask CpGrid about the aquifer cells, but I cannot find that this information is actually ever transferred to the load balanced grid. For the unbalanced grid the aquifer cells a added when processing the eclipse files it seems.

I didn't get a definite answer to this, but while working on a tangential issue I had a look at the `AquiferConfig`'s [`serializeOp()`](https://github.com/OPM/opm-common/blob/99e3608bf36deac109b1ecb3f42fe0ac543f6140/opm/input/eclipse/EclipseState/Aquifer/AquiferConfig.hpp#L83) member function.  That function will distribute all knowledge of numerical aquifers to all processes **provided** there are no changes to the numerical aquifer parts of `AquiferConfig` structure **after** `serializeOp()` [has been called](https://github.com/OPM/opm-simulators/blob/ecfe98dd3da96761917f8e9846dae4e7b12c2f75/opm/simulators/utils/readDeck.cpp#L559-L560).  That may of course still not be enough to get the information into the `CpGrid` on all ranks, but at least the information ought to be available everywhere.
